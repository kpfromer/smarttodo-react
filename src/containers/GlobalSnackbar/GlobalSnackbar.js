import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import Snackbar from '@material-ui/core/Snackbar';
import SpecialSnackbar from "../../components/SpecialSnackbar/SpecialSnackbar";
import { closeSnackbar } from "../../actions/snackbar";

export class GlobalSnackbar extends Component {

  static propTypes = {
    closeSnackbar: PropTypes.func.isRequired,
    isActive: PropTypes.bool.isRequired,
    message: PropTypes.string,
    variant: PropTypes.string,
    timing: PropTypes.number
  };

  render() {
    const { closeSnackbar, isActive, message, variant, timing } = this.props;
    return (
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center',
        }}
        open={isActive}
        autoHideDuration={timing}
        onClose={closeSnackbar}
      >
        <SpecialSnackbar
          onClose={closeSnackbar}
          variant={variant}
          message={message}
        />
      </Snackbar>
    );
  }
}

const mapStateToProps = state => {
  const { isActive, message, variant, timing } = state.snackbar;
  return {
    isActive,
    message,
    variant,
    timing
  }
};

export default connect(mapStateToProps, { closeSnackbar })(GlobalSnackbar);