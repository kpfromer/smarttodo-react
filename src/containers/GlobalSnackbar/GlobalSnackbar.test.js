import { GlobalSnackbar } from "./GlobalSnackbar";
import { SetupComponent } from "react-component-setup";
import Snackbar from '@material-ui/core/Snackbar';
import SpecialSnackbar from "../../components/SpecialSnackbar/SpecialSnackbar";

const { shallow: setup } = SetupComponent({
  component: GlobalSnackbar,
  defaultProps: {
    closeSnackbar: () => {},
    isActive: true,
    message: 'WORKING!',
    variant: 'success',
    timing: 5000
  }
});

describe('GlobalSnackbar', () => {
  let wrapper, mockCloseSnackbar;

  beforeEach(() => {
    mockCloseSnackbar = jest.fn();
    ({ wrapper } = setup({
      closeSnackbar: mockCloseSnackbar
    }));
  });

  it('renders without crashing', () => {
    setup();
  });

  describe('Snackbar', () => {
    let snackbar;

    beforeEach(() => {
      snackbar =  wrapper.find(Snackbar).props();
    });

    it('renders a snackbar', () => {
      expect(snackbar).toMatchSnapshot();
    });
    it('calls closeSnackbar on Snackbar close', () => {
      expect(snackbar.onClose).toBe(mockCloseSnackbar);
    });
  });

  describe('SpecialSnackbar', () => {
    let specialSnackbar;

    beforeEach(() => {
      specialSnackbar =  wrapper.find(SpecialSnackbar).props();
    });

    it('renders a SpecialSnackbar as content', () => {
      expect(specialSnackbar).toMatchSnapshot();
    });
    it('calls closeSnackbar on SpecialSnackbar close', () => {
      expect(specialSnackbar.onClose).toBe(mockCloseSnackbar);
    });
  });
});