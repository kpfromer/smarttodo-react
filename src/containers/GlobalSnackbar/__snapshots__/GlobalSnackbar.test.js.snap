// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSnackbar Snackbar renders a snackbar 1`] = `
Object {
  "anchorOrigin": Object {
    "horizontal": "center",
    "vertical": "bottom",
  },
  "autoHideDuration": 5000,
  "children": <WithStyles(SpecialSnackbar)
    message="WORKING!"
    onClose={[MockFunction]}
    variant="success"
  />,
  "onClose": [MockFunction],
  "open": true,
}
`;

exports[`GlobalSnackbar SpecialSnackbar renders a SpecialSnackbar as content 1`] = `
Object {
  "message": "WORKING!",
  "onClose": [MockFunction],
  "variant": "success",
}
`;
